src/exercise/dist/Main.js

```diff
@@ -0,0 +1,183 @@
+            // create delete button
+            task_section.appendChild(this.createDeleteBtn(task_section));
+        });
+    };
+
+    async addTask(task) {
+        if (task.value === ''){
+            task.setCustomValidity("You forgot to type the task!");
+            task.reportValidity();
+            return;
+        }
+        const loader = document.getElementsByClassName("loader")[0];
+        loader.style.display = 'block';
+        if (task.value) {
+            try{
+                await this.itemClient.addTask(task.value, new Date());
+            } catch(error) {
+                loader.style.display = 'none';
+                task.setCustomValidity(error.message);
+                task.reportValidity();
+                return;
+            }
+        }
+        loader.style.display = 'none';
+        await this.renderTasks();
+        task.value = '';
+    }
+
+    // returns the element that shows the date & time the task created
+    taskTime(dateString){
+        let date = new Date(dateString);
+        const p = document.createElement("p");
+        let minutes = String(date.getMinutes());
+        if (minutes.length === 1){
+            minutes = "0" + minutes;
+        }
+        p.innerHTML = date.getDate() + "/" + 
+                      (date.getMonth()+1) + "/" + 
+                      date.getFullYear() + "<br/>" + 
+                      date.getHours() + ":" + 
+                      minutes;
+        p.className = "time";
+        p.id = dateString;
+        p.style.fontSize = '10px';
+        return p;
+    };
+
+
+    // clear all tasks
+    clearAll(){
+        const taskSection = document.body.querySelectorAll("div.taskSection");
+        taskSection.forEach(elem => {
+            elem.classList.add("removed");
+            setTimeout(async () => {
+                elem.remove();
+                const task_date = elem.querySelectorAll('p');
+                await this.itemClient.deleteTask(task_date[0].innerText, task_date[1].id);
+            }, 800);
+        });
+    };
+
+
+    // sort tasks by date they created
+    sortTime (){
+        const taskSections = document.body.querySelectorAll("div.taskSection");
+        let ts_array = [].map.call(taskSections, (elem) => {return elem;});
+        ts_array.sort((a,b) => {
+            a = a.querySelector("div.tasks").querySelector("p.time").id;
+            b = b.querySelector("div.tasks").querySelector("p.time").id;
+            if (this.asc_time){
+                return new Date(a) - new Date(b);
+            }
+            return new Date(b) - new Date(a);
+        });
+        this.asc_time = !this.asc_time;
+        for (let i=0; i < ts_array.length; i++){
+            taskSections[i].parentNode.appendChild(ts_array[i]);
+        }
+    };
+
+
+    // sort tasks by name
+    sortAZ(){
+        const taskSections = document.body.querySelectorAll("div.taskSection");
+        let ts_array = [].map.call(taskSections, (elem) => {return elem;});
```

Comment:
What is the purpose of this line? creating a new array?
you can perform a [shallow copy](https://www.freecodecamp.org/news/how-to-clone-an-array-in-javascript-1d3183468f6a/) if you need to sort the array in place